src/Executor.cc

```diff
@@ -72,10 +72,21 @@ void Executor::cancel()
         return;
 
     comm_channel_->immediate_termination_requested_ = true;
+    while (const auto opt_msg = comm_channel_->queue_.try_pop());
     context_.variables = future_.get(); // Wait for thread to join
     comm_channel_->immediate_termination_requested_ = false;
 }
 
+void Executor::cancel(Sequence& sequence) {
+    if (not future_.valid())
+        return;
+    comm_channel_->immediate_termination_requested_ = true;
+    while(update(sequence));
+    if (future_.valid())
+        context_.variables = future_.get();
+    comm_channel_->immediate_termination_requested_ = false;
```

Comment:
Why do we set here the `immediate_termination_requested_ = false`? Thought that we wanted to abort the `Executor`? At least it needs some statements about swapping the state `immediate_termination_requested_` (I would forget about it in a few weeks :-/).

Reply:
At that point the execution has stopped already.

Setting it to `false` enabled reuse of the `comm`  object for the next run of something.
In principle we could also set it to `false` on `run_asynchroneously()`.
The question would be what is easier to comprehend?
I believe stuff that belongs to each other should be near in the code and not spread for apart.